codegen/end_to_end_test/test/operation/fragments_inline_nesting_test.dart

```diff
@@ -0,0 +1,41 @@
+import "package:test/test.dart";
+
+import 'package:end_to_end_test/fragments/__generated__/hero_with_interface_subtyped_fragments.data.gql.dart';
+
+void main() {
+  group("Fragments with InlineFragments for different sub-Types", () {
+    test('can deserialize correctly from sub-Typed data', () {
+      final jsonHuman = {
+        "__typename": "Human",
+        "id": "luke",
+        "name": "Luke",
+        "homePlanet": "Tattoine",
+        "birthday": 200059200000,
+        "friends": [
+          {
+            "__typename": "Droid",
+            "id": "r2d2",
+            "name": "R2-D2",
+            "primaryFunction": "Astromech"
+          },
+          {
+            "__typename": "Droid",
+            "id": "c3po",
+            "name": "C3-PO",
+            "primaryFunction": "Protocol Droid"
+          },
+        ],
+      };
+
+      final gqlHuman = GheroFieldsFragmentData__asHuman.fromJson(jsonHuman);
+
+      expect(gqlHuman?.id, equals("luke"));
+      expect(gqlHuman?.name, equals("Luke"));
+      expect(gqlHuman?.homePlanet, equals("Tattoine"));
+      expect(gqlHuman?.birthday, equals(DateTime.parse("1976-05-04T08:00:00")));
+      expect(gqlHuman?.friends?.first?.G__typename, equals("Droid"));
+      // FAILS: "The getter 'primaryFunction' isn't defined for the type 'GheroFieldsFragmentData__asHuman_friends'."
+      // expect(gqlHuman?.friends?.first?.primaryFunction, equals("Astromech"));
```

Comment:
Will be looking into it a bit deeper. 

Debugging into this, there are `Droid` specific fields in the `gqlHuman.friends` collection's items, BUT, those fields aren't accessible because the return type of the `friends` collection is for a narrower Type. (that is, it should be `_GheroFieldsFragmentData_asHuman_friends__asDroid`)

Reply:
I think this is expected behavour. Since it's possible that there are friends who are not droids but human, the return type has to be the base class with that has only the shared fields. To access the special fields, a type check or a cast is needed like

```dart
   final first = gqlHuman?.friends?.first as GheroFieldsFragmentData__asHuman_friends__asDroid;
   expect(a.primaryFunction, equals("Astromech"));
```

Reply:
Yep... I wasn't adding the type-filter... Just updated that!

Also extended this to what my original intent which was to demonstrate sub-type Inlines in the sub-Fragment.